docs/DeveloperGuide.md

```diff
@@ -1,38 +1,584 @@
 # Developer Guide
 
+---
```

Comment:
I think the sequence diagrams or any other diagrams theme can be the same for a more unified look